File: C#/326_Power_of_Three.cs
using System;

namespace LeetCode
{
    public class Solution
    {
        public bool IsPowerOfThree(int n)
        {
            if (n < 1)
                return false;

            while (n % 3 == 0)
            {
                n /= 3;
            }

            return n == 1;
        }

        public bool IsPowerOfThree2(int n)
        {
            return (Math.Log10(n) / Math.Log10(3)) % 1 == 0;
        }

        public bool IsPowerOfThree3(int n)
        {
            /*
             * 1162261467:
             *   Math.Pow(3, (int)(Math.Log(int.MaxValue, 3)))
             */
            return n > 0 && 1162261467 % n == 0;
        }

        //Testcase
        static void Main(string[] args)
        {
            Solution sol = new Solution();
            int rand_num = new Random().Next(0, 1000);
            if (sol.IsPowerOfThree(rand_num))
                Console.Write(rand_num + " is power of three.");
            else
                Console.Write(rand_num + " isn't power of three.");
        }
    }
}

File: Python/326_Power_of_Three.py
import math
import random


class Solution(object):
    def isPowerOfThree(self, n):
        """
        :type n: int
        :rtype: bool
        """
        if n < 1:
            return False

        while n % 3 == 0:
            n /= 3

        return n == 1

    def isPowerOfThree2(self, n):
        """
        :type n: int
        :rtype: bool
        """
        return n > 0 and 3 ** round(math.log(n, 3)) == n

    def isPowerOfThree3(self, n):
        """
        :type n: int
        :rtype: bool
        """
        #  1162261467:
        #    3 ** int(math.log(sys.maxint, 3))
        return n > 0 and 1162261467 % n == 0


def test():
    sol = Solution()
    rand_num = random.randint(0, 1000)

    if sol.isPowerOfThree(rand_num):
        print str(rand_num) + " is power of three."
    else:
        print str(rand_num) + " isn't power of three."


if __name__ == '__main__':
    test()
